sklearn/ensemble/base.py

```diff
@@ -178,3 +182,76 @@ def _partition_estimators(n_estimators, n_jobs):
     starts = np.cumsum(n_estimators_per_job)
 
     return n_jobs, n_estimators_per_job.tolist(), [0] + starts.tolist()
+
+
+class _BaseHeterogeneousEnsemble(MetaEstimatorMixin, _BaseComposition,
+                                 metaclass=ABCMeta):
+    """Base class for ensemble learners based on heterogeneous estimators."""
+    _required_parameters = ['estimators']
+
+    @property
+    def named_estimators(self):
+        return Bunch(**dict(self.estimators))
+
+    @abstractmethod
+    def __init__(self, estimators):
+        self.estimators = estimators
```

Comment:
Should we leave the setting of `self.estimators` for the child classes and make `_BaseHeterogeneousEnsemble` a mixin?

It mainly does input validation anyway?

No strong opinion, but I feel like setting a single attribute in the parent class makes reaading the code harder.

Reply:
Since we inherit from `_BaseComposition` and `BaseEstimator`, it will be weird to create a mixin which defines the `__init__`. Basically, by inheritance, it is already a base class and not a mixin.

Reply:
i'm suggesting to not have the init method